mergin/client.py

```diff
@@ -779,6 +780,23 @@ def get_file_diff(self, project_dir, file_path, output_diff, version_from, versi
             self, project_path, file_path, version_from=version_from, version_to=version_to, file_history=file_history
         )
         diffs = self.download_file_diffs(project_dir, file_path, versions_to_fetch[1:])
+        # if some changesets were cached before, they will be missed from the
```

Comment:
for the sake of clarity I would do following:
- make a comment here for 'download_file_diffs' like 'fetch versions files not already present in cache'
- ignore result of download function
- always parse file[history] to get all version files (at this point all should be cached) and check they actually exist
- sort in ascending order

Reply:
Done. Though if we compile list of changesets from the cached files there is no need to sort them, as versions are already sorted.